stout/flags.cc

```diff
@@ -186,7 +186,24 @@ void Parser::Parse(
           + "': missing value");
       continue;
     } else {
-      text = value.value();
+      const std::string str_value = value.value();
+      if (field->type() != google::protobuf::FieldDescriptor::TYPE_STRING) {
+        text = str_value;
+      } else {
+        size_t index_start = str_value.find_first_of("'\"");
+        size_t index_end = str_value.find_last_of("'\"");
+        if (index_start == std::string::npos
+            && index_end == std::string::npos) {
+          text = "'" + str_value + "'";
+        } else if (index_start == 0 && index_end == str_value.size() - 1) {
+          text = str_value;
+        } else {
+          errors.insert(
+              "String value ` " + str_value
+              + " ` should be framed by ' or \"");
+          continue;
+        }
```

Comment:
Reading through this PR I'm realizing that what ever `value.value()` is is exactly what we want to parse, regardless of what quotes it has, and all we need to do is _add_ double quotes to it to make sure that it's parseable by the text format parser. So this basically just becomes:
```suggestion
        text = "'" + value.value() + "'";
```

This is because the shell will remove the quotes as necessary. You can try this for yourself, write a simple C++ program that prints out each of `argv`. You'll see that on the shell if you do:

```
$ ./simple-program --arg1="hello world" --arg2=hello world
argv[0] = ./simple-program
argv[1] = --arg1=hello world
argv[2] = --arg2=hello
argv[3] = world
```
So what we actually care about is reintroducing the quotes so that `value.value()` is correct. In the tests below when we have something like `--foo='hello world'` what that actually means is that on the command line someone did `--foo="'hello world'"`. Does that make sense? We need the tests for when we just have `--foo=hello world` as well because that is what we'll get if someone on the command line does `--foo='hello world'`.

Reply:
Yep, I've already written simple program and saw this behaviour. Thank you for this catch!

Reply:
As you said when we have the case `--foo='hello'` it means that someone did `--foo="'hello'"` on the command line. In this case we should not do ` text = "'" + value.value() + "'";` because we will have the protobuf text-format parser error like the following:
```
Failed to parse flag 'foo' from normalized value '''hello''' due to protobuf text-format parser error(s).
```

Reply:
It means that we should check the value for existing quotes (' or ") .

Reply:
The case I'm trying to account for is when someone _does_ want to have a quote (single or double) in the argument that they are passing. For example, let's say someone does `--foo="'hello'" on the command line, that means that the value that they want for the argument is `'hello'` not `hello`.

Here's another example, if someone does `--foo="\"hello\""` on the command line that means that the string that they want is `"hello"` not `hello`. Same thing if they did `--foo='"hello"'`.

So my hope was that we could just use single quotes to wrap what ever we got from the command line so that the text-format parser would just do the right thing. That is, I was hoping if we have the text-format parser the string `'"hello"'` then it would treat the single quotes as indicating this is a string and then the double quotes would be part of the string so the result would be `"hello"`. I'm realizing now that this won't work for the case that you specified, but that's because `''hello''` is not a valid thing to parse! So I think we need to do two things, one is to wrap with `'` single quotes and the second is to escape any existing `'` single quotes that are already in the string! Can you try that?

Reply:
You will also need to escape backslashes, so if one the command line you got `--foo=hello\\\'` which will become for us `--foo=hello\'` we'll need to turn that _back_ into `hello\\\'`.

I think you can just use `absl:: CEscape()`! Then you can escape the string, and wrap it with `'` single quotes, and pass it on to the text-format parser.